test/e2e/framework/service/jig.go

```diff
@@ -474,7 +474,8 @@ func (j *TestJig) sanityCheckService(svc *v1.Service, svcType v1.ServiceType) (*
 	}
 
 	expectNodePorts := false
-	if svcType != v1.ServiceTypeClusterIP && svcType != v1.ServiceTypeExternalName {
+	if svcType != v1.ServiceTypeClusterIP && svcType != v1.ServiceTypeExternalName &&
```

Comment:
I would flip this.

```
if svcType == v1.ServiceTypeNodePort || (svcType == v1.ServiceTypeLoadBalancer && getAllocateLoadBalancerNodePorts(svc)) {
 //...
```

and

```
func getAllocateLoadBalancerNodePorts(svc *v1.Service) bool {
    if svc.Spec.AllocateLoadBalancerNodePorts == nil {
        return true // back-compat
    }
    return *svc.Spec.AllocateLoadBalancerNodePorts
}

Reply:
or even:

```
if needsNodePorts(svc) {
```

Those tiny helpers add a LOT to readability

Reply:
added helper func